### PR TITLE
chore: Bump vendor check go version to 1.18

### PR DESCRIPTION
**Pull request by alfrunes**
Changelog: None
Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

### CODE REVIEW

**Review by tranchitella — APPROVED** [review]

File: .gitlab-ci-check-golang-lint.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ test:static:
   needs: []
   except:
     - /^saas-[a-zA-Z0-9.]+$/
-  image: golangci/golangci-lint:v1.43.0
+  image: golangci/golangci-lint:v1.47.3
```
Comment (lluiscampos) [inline .gitlab-ci-check-golang-lint.yml:27]: I've been experimenting with this and the formatting will change enough to make release branches (and tags) of some repos don't pass anymore. We need an strategy on how to do this:  can we discuss it in the next QA meeting?

**Review by lluiscampos — CHANGES_REQUESTED** [review]

File: .gitlab-ci-check-golang-lint.yml (hunk 1 above)
Comment (alfrunes) [reply .gitlab-ci-check-golang-lint.yml:27]: Thanks. I'll hold back this PR until the release is done and we have a strategy for such changes. I will update the local CI configuration for the affected PR (mendersoftware/deployments#769).

**Comment by alfrunes** [comment]
@lluiscampos do you think we can merge this now that the release is over? Maybe we should even bump it to go1.19 while we're at it?

**Comment by lluiscampos** [comment]
> @lluiscampos do you think we can merge this now that the release is over? Maybe we should even bump it to go1.19 while we're at it?

We'll discuss it tomorrow in QA meeting.

Mender 3.4.0 is out but Kristian is working on 3.3.1 and I will shortly build 3.4.1, so the "release period" is not over unfortunately.

**Review by lluiscampos — APPROVED** [review]
As per team agreement in QA meeting, we can merge this one as-is and fix linting in the repos as we go. I have branched out (just before this PR) to `3.4.x` and `3.3.x` branches, which we can use on release branches for repos that turn hard to update.

**Pull request merged**